igbinary7 (php 7.1.0-dev) `make test` fails @ missing ps_globals symbol

php -v
        PHP 7.1.0-dev (cli) (built: Oct  4 2016 12:16:35) ( NTS )
        Copyright (c) 1997-2016 The PHP Group
        Zend Engine v3.1.0-dev, Copyright (c) 1998-2016 Zend Technologies
            with Zend OPcache v7.1.0-dev, Copyright (c) 1999-2016, by Zend Technologies

    cd igbinary
    git log | head
        commit 28b016149af74534a810b8c7b6a840f5a0522df6
        Author: Tyson Andre <tyson.andre@uwaterloo.ca>
        Date:   Tue Oct 4 08:17:05 2016 -0700

            Fixes the php7 APCU serialization/unserialization handler (#65)

            Fixes https://github.com/igbinary/igbinary7/issues/21

            There is a bug in the APCU code for custom serializers'
            serialization/unserialization methods.

    phpize
        Configuring for:
        PHP Api Version:         20160303
        Zend Module Api No:      20160303
        Zend Extension Api No:   320160303

    ./configure \
     --prefix=/usr/local \
     --enable-shared --disable-static \
     --enable-igbinary \
     --with-php-config=/usr/local/php7/bin/php-config
        ...
        checking PHP version... PHP 7
        ...

    make -j1
        ...
        Build complete.
        Don't forget to run 'make test'.

    make V=1 test
        Build complete.
        Don't forget to run 'make test'.

        PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/src/igbinary/modules/igbinary.so' - /usr/local/src/igbinary/modules/igbinary.so: undefined symbol: ps_globals in Unknown on line 0
        PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/src/igbinary/modules/igbinary.so' - /usr/local/src/igbinary/modules/igbinary.so: undefined symbol: ps_globals in Unknown on line 0

        =====================================================================
        PHP         : /usr/local/php7/bin/php 
        Warning: PHP Startup: Unable to load dynamic library '/usr/local/src/igbinary/modules/igbinary.so' - /usr/local/src/igbinary/modules/igbinary.so: undefined symbol: ps_globals in Unknown on line 0

        PHP_SAPI    : cli
        PHP_VERSION : 7.1.0-dev
        ZEND_VERSION: 3.1.0-dev
        PHP_OS      : Linux - Linux dev.loc 4.7.6-1.gfb37fcc-default #1 SMP PREEMPT Fri Sep 30 12:22:14 UTC 2016 (fb37fcc) x86_64
        INI actual  : /usr/local/src/igbinary/tmp-php.ini
        More .INIs  :   
        CWD         : /usr/local/src/igbinary
        Extra dirs  : 
        VALGRIND    : Not used
        =====================================================================
        TIME START 2016-10-04 14:00:33
        =====================================================================
        SKIP Check for igbinary presence [tests/igbinary_001.phpt] 
        FAIL Check for null serialisation [tests/igbinary_002.phpt] 
        FAIL Check for bool serialisation [tests/igbinary_003.phpt] 
        FAIL Check for integer serialisation [tests/igbinary_004.phpt] 
        FAIL Check for double serialisation [tests/igbinary_005.phpt] 
        FAIL Check for simple string serialization [tests/igbinary_006.phpt] 
        FAIL Check for simple array serialization [tests/igbinary_007.phpt] 
        FAIL Check for array+string serialization [tests/igbinary_008.phpt] 
        SKIP Check for reference serialisation [tests/igbinary_009.phpt] reason: no igbinary
        SKIP Check for reference serialization (Original example, not using var_dump) [tests/igbinary_009b.phpt] reason: no igbinary
        FAIL Array test [tests/igbinary_010.phpt] 
        FAIL Object test [tests/igbinary_012.phpt] 
        FAIL Object-Array test [tests/igbinary_013.phpt] 
        FAIL Object-Reference test [tests/igbinary_014.phpt] 
        SKIP Check for serialization handler [tests/igbinary_015.phpt] reason: session extension not loaded
        SKIP Check for serialization handler, ini-directive [tests/igbinary_015b.phpt] reason: session extension not loaded
        FAIL Object test, __sleep [tests/igbinary_016.phpt] 
        FAIL Object test, __wakeup [tests/igbinary_017.phpt] 
        FAIL Object test, __sleep error cases [tests/igbinary_018.phpt] 
        FAIL Object test, __autoload [tests/igbinary_019.phpt] 
        FAIL Object test, incomplete class [tests/igbinary_020.phpt] 
        FAIL Object Serializable interface [tests/igbinary_021.phpt] 
        FAIL Object test, unserialize_callback_func [tests/igbinary_022.phpt] 
        SKIP Resource [tests/igbinary_023.phpt] reason: extenion not loaded
        FAIL Recursive objects [tests/igbinary_024.phpt] 
        FAIL Object test, array of objects with __sleep [tests/igbinary_025.phpt] 
        SKIP Cyclic array test [tests/igbinary_026.phpt] reason: no igbinary
        SKIP Cyclic array test 2 [tests/igbinary_026b.phpt] reason: no igbinary
        SKIP Check for serialization handler [tests/igbinary_027.phpt] reason: session extension not loaded
        SKIP Serialize object into session, full set [tests/igbinary_028.phpt] reason: session extension not loaded
        SKIP Igbinary module info [tests/igbinary_029.phpt] 
        SKIP Unserialize invalid data [tests/igbinary_030.phpt] reason: no igbinary
        FAIL Object Serializable interface throws exceptions [tests/igbinary_031.phpt] 
        SKIP Object test, __sleep and __wakeup exceptions [tests/igbinary_032.phpt] reason: no igbinary
        SKIP Object test, cyclic references [tests/igbinary_033.phpt] reason: no igbinary
        SKIP Unserialize invalid random data [tests/igbinary_034.phpt] reason: no igbinary
        FAIL b0rked random data test [tests/igbinary_040.phpt] 
        FAIL Check for double NaN, Inf, -Inf, 0, and -0. IEEE 754 doubles [tests/igbinary_041.phpt] 
        SKIP Closure [tests/igbinary_042.phpt] reason: no igbinary
        SKIP Object serialization with compact strings [tests/igbinary_043.phpt] 
        FAIL Check for double extremes [tests/igbinary_044.phpt] 
        SKIP APC serializer registration [tests/igbinary_045.phpt] reason: APC not loaded
        SKIP APCu serializer registration [tests/igbinary_045b.phpt] reason: APCu not loaded
        SKIP APCu serializer registration - more data types [tests/igbinary_045c.phpt] reason: APCu not loaded
        FAIL Correctly unserialize scalar refs. [tests/igbinary_046.phpt] 
        FAIL Correctly unserialize multiple object refs. [tests/igbinary_046b.phpt] 
        FAIL Correctly unserialize multiple array refs. [tests/igbinary_046c.phpt] 
        FAIL Correctly unserialize multiple object refs and non-refs. [tests/igbinary_046d.phpt] 
        SKIP Check for serialization handler, SessionHandlerInterface [tests/igbinary_047.phpt] reason: session extension not loaded
        SKIP Object test, __set not called for private attr in extended class [tests/igbinary_048.phpt] reason: no igbinary
        FAIL Object test, __set not called for private attr in extended class [tests/igbinary_048b.phpt] 
        FAIL Correctly unserialize multiple references in arrays [tests/igbinary_049.phpt] 
        FAIL Correctly unserialize multiple references in objects [tests/igbinary_049b.phpt] 
        FAIL Correctly unserialize cyclic object references [tests/igbinary_050.phpt] 
        FAIL Object test, __wakeup (With multiple references) [tests/igbinary_051.phpt] 
        FAIL Object Serializable interface can be serialized in references [tests/igbinary_052.phpt] 
        FAIL __wakeup can modify properties without affecting other objects [tests/igbinary_053.phpt] 
        FAIL __wakeup can add dynamic properties without affecting other objects [tests/igbinary_054.phpt] 
        FAIL __wakeup can replace a copy of the object referring to the root node. [tests/igbinary_055.phpt] 
        SKIP Nested objects cause segfault, php bug #54662 [tests/igbinary_bug54662.phpt] 
        FAIL igbinary_unserialize causes segfault on 3rd call for objects with dynamic property [tests/igbinary_bug72134.phpt] 
        FAIL Unserialize backwards compatible with v1. [tests/igbinary_unserialize_v1_compatible.phpt] 
        =====================================================================
        TIME END 2016-10-04 14:00:34

        =====================================================================
        TEST RESULT SUMMARY
        ---------------------------------------------------------------------
        Exts skipped    :    0
        Exts tested     :   10
        ---------------------------------------------------------------------

        Number of tests :   62                39
        Tests skipped   :   23 ( 37.1%) --------
        Tests warned    :    0 (  0.0%) (  0.0%)
        Tests failed    :   39 ( 62.9%) (100.0%)
        Expected fail   :    0 (  0.0%) (  0.0%)
        Tests passed    :    0 (  0.0%) (  0.0%)
        ---------------------------------------------------------------------
        Time taken      :    1 seconds
        =====================================================================

        =====================================================================
        FAILED TEST SUMMARY
        ---------------------------------------------------------------------
        Check for null serialisation [tests/igbinary_002.phpt]
        Check for bool serialisation [tests/igbinary_003.phpt]
        Check for integer serialisation [tests/igbinary_004.phpt]
        Check for double serialisation [tests/igbinary_005.phpt]
        Check for simple string serialization [tests/igbinary_006.phpt]
        Check for simple array serialization [tests/igbinary_007.phpt]
        Check for array+string serialization [tests/igbinary_008.phpt]
        Array test [tests/igbinary_010.phpt]
        Object test [tests/igbinary_012.phpt]
        Object-Array test [tests/igbinary_013.phpt]
        Object-Reference test [tests/igbinary_014.phpt]
        Object test, __sleep [tests/igbinary_016.phpt]
        Object test, __wakeup [tests/igbinary_017.phpt]
        Object test, __sleep error cases [tests/igbinary_018.phpt]
        Object test, __autoload [tests/igbinary_019.phpt]
        Object test, incomplete class [tests/igbinary_020.phpt]
        Object Serializable interface [tests/igbinary_021.phpt]
        Object test, unserialize_callback_func [tests/igbinary_022.phpt]
        Recursive objects [tests/igbinary_024.phpt]
        Object test, array of objects with __sleep [tests/igbinary_025.phpt]
        Object Serializable interface throws exceptions [tests/igbinary_031.phpt]
        b0rked random data test [tests/igbinary_040.phpt]
        Check for double NaN, Inf, -Inf, 0, and -0. IEEE 754 doubles [tests/igbinary_041.phpt]
        Check for double extremes [tests/igbinary_044.phpt]
        Correctly unserialize scalar refs. [tests/igbinary_046.phpt]
        Correctly unserialize multiple object refs. [tests/igbinary_046b.phpt]
        Correctly unserialize multiple array refs. [tests/igbinary_046c.phpt]
        Correctly unserialize multiple object refs and non-refs. [tests/igbinary_046d.phpt]
        Object test, __set not called for private attr in extended class [tests/igbinary_048b.phpt]
        Correctly unserialize multiple references in arrays [tests/igbinary_049.phpt]
        Correctly unserialize multiple references in objects [tests/igbinary_049b.phpt]
        Correctly unserialize cyclic object references [tests/igbinary_050.phpt]
        Object test, __wakeup (With multiple references) [tests/igbinary_051.phpt]
        Object Serializable interface can be serialized in references [tests/igbinary_052.phpt]
        __wakeup can modify properties without affecting other objects [tests/igbinary_053.phpt]
        __wakeup can add dynamic properties without affecting other objects [tests/igbinary_054.phpt]
        __wakeup can replace a copy of the object referring to the root node. [tests/igbinary_055.phpt]
        igbinary_unserialize causes segfault on 3rd call for objects with dynamic property [tests/igbinary_bug72134.phpt]
        Unserialize backwards compatible with v1. [tests/igbinary_unserialize_v1_compatible.phpt]
        =====================================================================

        You may have found a problem in PHP.
        This report can be automatically sent to the PHP QA team at
        http://qa.php.net/reports and http://news.php.net/php.qa.reports
        This gives us a better understanding of PHP's behavior.
        If you don't want to send the report immediately you can choose
        option "s" to save it.  You can then email it to qa-reports@lists.php.net later.

    Checking

        nm /usr/local/src/igbinary/modules/igbinary.so | grep ps_global
            U ps_globals

        nm $(/usr/local/php7/bin/php-config --extension-dir)/session.so | grep ps_global
            0000000000216b00 B ps_globals

Taking a look at verbose `make` as well, no *errors* but a bunch of non-fatal warnings ...

    make clean
    make V=1 test
        /bin/sh /usr/local/src/igbinary/libtool --mode=compile /usr/bin/gcc-6 -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/usr/local/src/igbinary -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native   -c /usr/local/src/igbinary/src/php7/igbinary.c -o src/php7/igbinary.lo 
        mkdir src/php7/.libs
         /usr/bin/gcc-6 -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/usr/local/src/igbinary -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -c /usr/local/src/igbinary/src/php7/igbinary.c  -fPIC -DPIC -o src/php7/.libs/igbinary.o
        In file included from /usr/local/php7/include/php/Zend/zend_modules.h:26:0,
                         from /usr/local/php7/include/php/Zend/zend_API.h:27,
                         from /usr/local/php7/include/php/main/php.h:39,
                         from /usr/local/src/igbinary/src/php7/igbinary.c:18:
        /usr/local/php7/include/php/Zend/zend_compile.h:758:1: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
         ZEND_API void zend_try_exception_handler();
         ^~~~~~~~
        In file included from /usr/local/src/igbinary/src/php7/igbinary.c:18:0:
        /usr/local/php7/include/php/main/php.h:313:46: warning: function declaration isn’t a prototype [-Wstrict-prototypes]
         static inline ZEND_ATTRIBUTE_DEPRECATED void php_std_error_handling() {}
                                                      ^~~~~~~~~~~~~~~~~~~~~~
        In file included from /usr/local/php7/include/php/main/php.h:39:0,
                         from /usr/local/src/igbinary/src/php7/igbinary.c:18:
        /usr/local/php7/include/php/Zend/zend_API.h:144:35: warning: no previous prototype for ‘get_module’ [-Wmissing-prototypes]
          ZEND_DLEXPORT zend_module_entry *get_module(void) { return &name##_module_entry; }\
                                           ^
        /usr/local/src/igbinary/src/php7/igbinary.c:287:1: note: in expansion of macro ‘ZEND_GET_MODULE’
         ZEND_GET_MODULE(igbinary)
         ^~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘php_igbinary_init_globals’:
        /usr/local/src/igbinary/src/php7/igbinary.c:298:62: warning: declaration of ‘igbinary_globals’ shadows a global declaration [-Wshadow]
         static void php_igbinary_init_globals(zend_igbinary_globals *igbinary_globals) {
                                                                      ^~~~~~~~~~~~~~~~
        In file included from /usr/local/php7/include/php/main/php.h:39:0,
                         from /usr/local/src/igbinary/src/php7/igbinary.c:18:
        /usr/local/src/igbinary/src/php7/igbinary.c:283:29: note: shadowed declaration is here
         ZEND_DECLARE_MODULE_GLOBALS(igbinary)
                                     ^
        /usr/local/php7/include/php/Zend/zend_API.h:170:31: note: in definition of macro ‘ZEND_DECLARE_MODULE_GLOBALS’
          zend_##module_name##_globals module_name##_globals;
                                       ^~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘ps_srlzr_encode_igbinary’:
        /usr/local/src/igbinary/src/php7/igbinary.c:588:28: warning: pointer targets in passing argument 1 of ‘zend_string_init’ differ in signedness [-Wpointer-sign]
          result = zend_string_init(igsd.buffer, igsd.buffer_size, 0);
                                    ^~~~
        In file included from /usr/local/php7/include/php/Zend/zend.h:35:0,
                         from /usr/local/php7/include/php/main/php.h:35,
                         from /usr/local/src/igbinary/src/php7/igbinary.c:18:
        /usr/local/php7/include/php/Zend/zend_string.h:156:40: note: expected ‘const char *’ but argument is of type ‘uint8_t * {aka unsigned char *}’
         static zend_always_inline zend_string *zend_string_init(const char *str, size_t len, int persistent)
                                                ^~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:567:11: warning: unused variable ‘tmpbuf’ [-Wunused-variable]
          uint8_t *tmpbuf;
                   ^~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:565:15: warning: unused variable ‘key’ [-Wunused-variable]
          zend_string *key;
                       ^~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘ps_srlzr_decode_igbinary’:
        /usr/local/src/igbinary/src/php7/igbinary.c:597:6: warning: unused variable ‘tmp_int’ [-Wunused-variable]
          int tmp_int;
              ^~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘igbinary_unserialize_array’:
        /usr/local/src/igbinary/src/php7/igbinary.c:2079:9: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
             key = "";
                 ^
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘igbinary_serialize_zval’:
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1462:9: note: called from here
             if (igbinary_serialize8(igsd, (uint8_t) igbinary_type_object_ser8 TSRMLS_CC) != 0) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1469:9: note: called from here
             if (igbinary_serialize8(igsd, (uint8_t) serialized_len TSRMLS_CC) != 0) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1476:9: note: called from here
             if (igbinary_serialize8(igsd, (uint8_t) igbinary_type_object_ser16 TSRMLS_CC) != 0) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1490:9: note: called from here
             if (igbinary_serialize8(igsd, (uint8_t) igbinary_type_object_ser32 TSRMLS_CC) != 0) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1219:7: note: called from here
           if (igbinary_serialize8(igsd, igbinary_type_array8 TSRMLS_CC) != 0) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1223:7: note: called from here
           if (igbinary_serialize8(igsd, n TSRMLS_CC) != 0) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1227:7: note: called from here
           if (igbinary_serialize8(igsd, igbinary_type_array16 TSRMLS_CC) != 0) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1235:7: note: called from here
           if (igbinary_serialize8(igsd, igbinary_type_array32 TSRMLS_CC) != 0) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:817:9: note: called from here
          return igbinary_serialize8(igsd, igbinary_type_null TSRMLS_CC);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:925:19: warning: inlining failed in call to ‘igbinary_serialize_string’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize_string(struct igbinary_serialize_data *igsd, char *s, size_t len TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1271:9: note: called from here
             if (igbinary_serialize_string(igsd, ZSTR_VAL(prop_name), ZSTR_LEN(prop_name) TSRMLS_CC) != 0) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:925:19: warning: inlining failed in call to ‘igbinary_serialize_string’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize_string(struct igbinary_serialize_data *igsd, char *s, size_t len TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1306:11: note: called from here
               if (igbinary_serialize_string(igsd, Z_STRVAL_P(d), Z_STRLEN_P(d) TSRMLS_CC) != 0) {
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:817:9: note: called from here
          return igbinary_serialize8(igsd, igbinary_type_null TSRMLS_CC);
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:925:19: warning: inlining failed in call to ‘igbinary_serialize_string’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize_string(struct igbinary_serialize_data *igsd, char *s, size_t len TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1321:10: note: called from here
              if (igbinary_serialize_string(igsd, ZSTR_VAL(mangled_prop_name), ZSTR_LEN(mangled_prop_name) TSRMLS_CC) != 0) {
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:925:19: warning: inlining failed in call to ‘igbinary_serialize_string’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize_string(struct igbinary_serialize_data *igsd, char *s, size_t len TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1334:9: note: called from here
             if (igbinary_serialize_string(igsd, Z_STRVAL_P(d), Z_STRLEN_P(d) TSRMLS_CC) != 0) {
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1562:7: note: called from here
             r = igbinary_serialize8(igsd, igbinary_type_array8 TSRMLS_CC);
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:758:19: warning: inlining failed in call to ‘igbinary_serialize8’: call is unlikely and code size would grow [-Winline]
         inline static int igbinary_serialize8(struct igbinary_serialize_data *igsd, uint8_t i TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1564:8: note: called from here
              r = igbinary_serialize8(igsd, 0 TSRMLS_CC);
              ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘ps_srlzr_encode_igbinary’:
        /usr/local/src/igbinary/src/php7/igbinary.c:721:20: warning: inlining failed in call to ‘igbinary_serialize_data_deinit.constprop’: call is unlikely and code size would grow [-Winline]
         inline static void igbinary_serialize_data_deinit(struct igbinary_serialize_data *igsd, int free_buffer TSRMLS_DC) {
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:576:3: note: called from here
           igbinary_serialize_data_deinit(&igsd, 1 TSRMLS_CC);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:721:20: warning: inlining failed in call to ‘igbinary_serialize_data_deinit.constprop’: call is unlikely and code size would grow [-Winline]
         inline static void igbinary_serialize_data_deinit(struct igbinary_serialize_data *igsd, int free_buffer TSRMLS_DC) {
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:581:3: note: called from here
           igbinary_serialize_data_deinit(&igsd, 1 TSRMLS_CC);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘igbinary_unserialize_zval’:
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘ps_srlzr_decode_igbinary’:
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘igbinary_serialize_ex’:
        /usr/local/src/igbinary/src/php7/igbinary.c:721:20: warning: inlining failed in call to ‘igbinary_serialize_data_deinit.constprop’: call is unlikely and code size would grow [-Winline]
         inline static void igbinary_serialize_data_deinit(struct igbinary_serialize_data *igsd, int free_buffer TSRMLS_DC) {
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:463:3: note: called from here
           igbinary_serialize_data_deinit(&igsd, 1 TSRMLS_CC);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c: In function ‘igbinary_unserialize’:
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:1960:19: warning: inlining failed in call to ‘igbinary_unserialize_array’: --param max-inline-insns-single limit reached [-Winline]
         inline static int igbinary_unserialize_array(struct igbinary_unserialize_data *igsd, enum igbinary_type t, zval *const z, int flags TSRMLS_DC) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~
        /usr/local/src/igbinary/src/php7/igbinary.c:2301:6: note: called from here
            r = igbinary_unserialize_array(igsd, t, IGB_REF_VAL(igsd, ref_n), flags|WANT_OBJECT TSRMLS_CC);
            ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        /bin/sh /usr/local/src/igbinary/libtool --mode=compile /usr/bin/gcc-6 -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/usr/local/src/igbinary -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native   -c /usr/local/src/igbinary/src/php7/hash_si.c -o src/php7/hash_si.lo 
         /usr/bin/gcc-6 -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/usr/local/src/igbinary -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -c /usr/local/src/igbinary/src/php7/hash_si.c  -fPIC -DPIC -o src/php7/.libs/hash_si.o
        /bin/sh /usr/local/src/igbinary/libtool --mode=compile /usr/bin/gcc-6 -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/usr/local/src/igbinary -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native   -c /usr/local/src/igbinary/src/php7/hash_si_ptr.c -o src/php7/hash_si_ptr.lo 
         /usr/bin/gcc-6 -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000 -I. -I/usr/local/src/igbinary -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -c /usr/local/src/igbinary/src/php7/hash_si_ptr.c  -fPIC -DPIC -o src/php7/.libs/hash_si_ptr.o
        /usr/local/src/igbinary/src/php7/hash_si_ptr.c: In function ‘hash_si_ptr_deinit’:
        /usr/local/src/igbinary/src/php7/hash_si_ptr.c:74:9: warning: unused variable ‘i’ [-Wunused-variable]
          size_t i;
                 ^
        /bin/sh /usr/local/src/igbinary/libtool --mode=link /usr/bin/gcc-6 -DPHP_ATOM_INC -I/usr/local/src/igbinary/include -I/usr/local/src/igbinary/main -I/usr/local/src/igbinary -I/usr/local/php7/include/php -I/usr/local/php7/include/php/main -I/usr/local/php7/include/php/TSRM -I/usr/local/php7/include/php/Zend -I/usr/local/php7/include/php/ext -I/usr/local/php7/include/php/ext/date/lib  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native -D_FORTIFY_SOURCE=2 -DHAVE_CONFIG_H  -O3 -Wall -fstack-protector -funwind-tables -fasynchronous-unwind-tables -fmessage-length=0 -grecord-gcc-switches -march=native -mtune=native   -o igbinary.la -export-dynamic -avoid-version -prefer-pic -module -rpath /usr/local/src/igbinary/modules  src/php7/igbinary.lo src/php7/hash_si.lo src/php7/hash_si_ptr.lo 
        mkdir .libs
        /usr/bin/gcc-6 -shared  src/php7/.libs/igbinary.o src/php7/.libs/hash_si.o src/php7/.libs/hash_si_ptr.o   -march=native -mtune=native -march=native -mtune=native -Wl,-soname -Wl,igbinary.so -o .libs/igbinary.so
        creating igbinary.la
        (cd .libs && rm -f igbinary.la && ln -s ../igbinary.la igbinary.la)
        /bin/sh /usr/local/src/igbinary/libtool --mode=install cp ./igbinary.la /usr/local/src/igbinary/modules
        cp ./.libs/igbinary.so /usr/local/src/igbinary/modules/igbinary.so
        cp ./.libs/igbinary.lai /usr/local/src/igbinary/modules/igbinary.la
        PATH="$PATH:/sbin" ldconfig -n /usr/local/src/igbinary/modules
        ----------------------------------------------------------------------
        Libraries have been installed in:
           /usr/local/src/igbinary/modules

        If you ever happen to want to link against installed libraries
        in a given directory, LIBDIR, you must either use libtool, and
        specify the full pathname of the library, or use the `-LLIBDIR'
        flag during linking and do at least one of the following:
           - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
             during execution
           - add LIBDIR to the `LD_RUN_PATH' environment variable
             during linking
           - use the `-Wl,--rpath -Wl,LIBDIR' linker flag
           - have your system administrator add LIBDIR to `/etc/ld.so.conf'

        See any operating system documentation about shared libraries for
        more information, such as the ld(1) and ld.so(8) manual pages.
        ----------------------------------------------------------------------

        Build complete.
        Don't forget to run 'make test'.

        PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/src/igbinary/modules/igbinary.so' - /usr/local/src/igbinary/modules/igbinary.so: undefined symbol: ps_globals in Unknown on line 0
        PHP Warning:  PHP Startup: Unable to load dynamic library '/usr/local/src/igbinary/modules/igbinary.so' - /usr/local/src/igbinary/modules/igbinary.so: undefined symbol: ps_globals in Unknown on line 0

        =====================================================================
        PHP         : /usr/local/php7/bin/php 
        Warning: PHP Startup: Unable to load dynamic library '/usr/local/src/igbinary/modules/igbinary.so' - /usr/local/src/igbinary/modules/igbinary.so: undefined symbol: ps_globals in Unknown on line 0

        PHP_SAPI    : cli
        PHP_VERSION : 7.1.0-dev
        ZEND_VERSION: 3.1.0-dev
        PHP_OS      : Linux - Linux deskEdgar.presence-group.net 4.8.0-2.g55165a4-default #1 SMP PREEMPT Mon Oct 3 14:15:10 UTC 2016 (55165a4) x86_64
        INI actual  : /usr/local/src/igbinary/tmp-php.ini
        More .INIs  :   
        CWD         : /usr/local/src/igbinary
        Extra dirs  : 
        VALGRIND    : Not used
        =====================================================================
        TIME START 2016-10-05 06:46:06
        =====================================================================
        SKIP Check for igbinary presence [tests/igbinary_001.phpt] 
        FAIL Check for null serialisation [tests/igbinary_002.phpt] 
        FAIL Check for bool serialisation [tests/igbinary_003.phpt] 
        FAIL Check for integer serialisation [tests/igbinary_004.phpt] 
        FAIL Check for double serialisation [tests/igbinary_005.phpt] 
        FAIL Check for simple string serialization [tests/igbinary_006.phpt] 
        FAIL Check for simple array serialization [tests/igbinary_007.phpt] 
        FAIL Check for array+string serialization [tests/igbinary_008.phpt] 
        SKIP Check for reference serialisation [tests/igbinary_009.phpt] reason: no igbinary
        SKIP Check for reference serialization (Original example, not using var_dump) [tests/igbinary_009b.phpt] reason: no igbinary
        FAIL Array test [tests/igbinary_010.phpt] 
        FAIL Object test [tests/igbinary_012.phpt] 
        FAIL Object-Array test [tests/igbinary_013.phpt] 
        FAIL Object-Reference test [tests/igbinary_014.phpt] 
        SKIP Check for serialization handler [tests/igbinary_015.phpt] reason: session extension not loaded
        SKIP Check for serialization handler, ini-directive [tests/igbinary_015b.phpt] reason: session extension not loaded
        FAIL Object test, __sleep [tests/igbinary_016.phpt] 
        FAIL Object test, __wakeup [tests/igbinary_017.phpt] 
        FAIL Object test, __sleep error cases [tests/igbinary_018.phpt] 
        FAIL Object test, __autoload [tests/igbinary_019.phpt] 
        FAIL Object test, incomplete class [tests/igbinary_020.phpt] 
        FAIL Object Serializable interface [tests/igbinary_021.phpt] 
        FAIL Object test, unserialize_callback_func [tests/igbinary_022.phpt] 
        SKIP Resource [tests/igbinary_023.phpt] reason: extenion not loaded
        FAIL Recursive objects [tests/igbinary_024.phpt] 
        FAIL Object test, array of objects with __sleep [tests/igbinary_025.phpt] 
        SKIP Cyclic array test [tests/igbinary_026.phpt] reason: no igbinary
        SKIP Cyclic array test 2 [tests/igbinary_026b.phpt] reason: no igbinary
        SKIP Check for serialization handler [tests/igbinary_027.phpt] reason: session extension not loaded
        SKIP Serialize object into session, full set [tests/igbinary_028.phpt] reason: session extension not loaded
        SKIP Igbinary module info [tests/igbinary_029.phpt] 
        SKIP Unserialize invalid data [tests/igbinary_030.phpt] reason: no igbinary
        FAIL Object Serializable interface throws exceptions [tests/igbinary_031.phpt] 
        SKIP Object test, __sleep and __wakeup exceptions [tests/igbinary_032.phpt] reason: no igbinary
        SKIP Object test, cyclic references [tests/igbinary_033.phpt] reason: no igbinary
        SKIP Unserialize invalid random data [tests/igbinary_034.phpt] reason: no igbinary
        FAIL b0rked random data test [tests/igbinary_040.phpt] 
        FAIL Check for double NaN, Inf, -Inf, 0, and -0. IEEE 754 doubles [tests/igbinary_041.phpt] 
        SKIP Closure [tests/igbinary_042.phpt] reason: no igbinary
        SKIP Object serialization with compact strings [tests/igbinary_043.phpt] 
        FAIL Check for double extremes [tests/igbinary_044.phpt] 
        SKIP APC serializer registration [tests/igbinary_045.phpt] reason: APC not loaded
        SKIP APCu serializer registration [tests/igbinary_045b.phpt] reason: APCu not loaded
        SKIP APCu serializer registration - more data types [tests/igbinary_045c.phpt] reason: APCu not loaded
        FAIL Correctly unserialize scalar refs. [tests/igbinary_046.phpt] 
        FAIL Correctly unserialize multiple object refs. [tests/igbinary_046b.phpt] 
        FAIL Correctly unserialize multiple array refs. [tests/igbinary_046c.phpt] 
        FAIL Correctly unserialize multiple object refs and non-refs. [tests/igbinary_046d.phpt] 
        SKIP Check for serialization handler, SessionHandlerInterface [tests/igbinary_047.phpt] reason: session extension not loaded
        SKIP Object test, __set not called for private attr in extended class [tests/igbinary_048.phpt] reason: no igbinary
        FAIL Object test, __set not called for private attr in extended class [tests/igbinary_048b.phpt] 
        FAIL Correctly unserialize multiple references in arrays [tests/igbinary_049.phpt] 
        FAIL Correctly unserialize multiple references in objects [tests/igbinary_049b.phpt] 
        FAIL Correctly unserialize cyclic object references [tests/igbinary_050.phpt] 
        FAIL Object test, __wakeup (With multiple references) [tests/igbinary_051.phpt] 
        FAIL Object Serializable interface can be serialized in references [tests/igbinary_052.phpt] 
        FAIL __wakeup can modify properties without affecting other objects [tests/igbinary_053.phpt] 
        FAIL __wakeup can add dynamic properties without affecting other objects [tests/igbinary_054.phpt] 
        FAIL __wakeup can replace a copy of the object referring to the root node. [tests/igbinary_055.phpt] 
        SKIP Nested objects cause segfault, php bug #54662 [tests/igbinary_bug54662.phpt] 
        FAIL igbinary_unserialize causes segfault on 3rd call for objects with dynamic property [tests/igbinary_bug72134.phpt] 
        FAIL Unserialize backwards compatible with v1. [tests/igbinary_unserialize_v1_compatible.phpt] 
        =====================================================================
        TIME END 2016-10-05 06:46:07

        =====================================================================
        TEST RESULT SUMMARY
        ---------------------------------------------------------------------
        Exts skipped    :    0
        Exts tested     :   10
        ---------------------------------------------------------------------

        Number of tests :   62                39
        Tests skipped   :   23 ( 37.1%) --------
        Tests warned    :    0 (  0.0%) (  0.0%)
        Tests failed    :   39 ( 62.9%) (100.0%)
        Expected fail   :    0 (  0.0%) (  0.0%)
        Tests passed    :    0 (  0.0%) (  0.0%)
        ---------------------------------------------------------------------
        Time taken      :    1 seconds
        =====================================================================

        =====================================================================
        FAILED TEST SUMMARY
        ---------------------------------------------------------------------
        Check for null serialisation [tests/igbinary_002.phpt]
        Check for bool serialisation [tests/igbinary_003.phpt]
        Check for integer serialisation [tests/igbinary_004.phpt]
        Check for double serialisation [tests/igbinary_005.phpt]
        Check for simple string serialization [tests/igbinary_006.phpt]
        Check for simple array serialization [tests/igbinary_007.phpt]
        Check for array+string serialization [tests/igbinary_008.phpt]
        Array test [tests/igbinary_010.phpt]
        Object test [tests/igbinary_012.phpt]
        Object-Array test [tests/igbinary_013.phpt]
        Object-Reference test [tests/igbinary_014.phpt]
        Object test, __sleep [tests/igbinary_016.phpt]
        Object test, __wakeup [tests/igbinary_017.phpt]
        Object test, __sleep error cases [tests/igbinary_018.phpt]
        Object test, __autoload [tests/igbinary_019.phpt]
        Object test, incomplete class [tests/igbinary_020.phpt]
        Object Serializable interface [tests/igbinary_021.phpt]
        Object test, unserialize_callback_func [tests/igbinary_022.phpt]
        Recursive objects [tests/igbinary_024.phpt]
        Object test, array of objects with __sleep [tests/igbinary_025.phpt]
        Object Serializable interface throws exceptions [tests/igbinary_031.phpt]
        b0rked random data test [tests/igbinary_040.phpt]
        Check for double NaN, Inf, -Inf, 0, and -0. IEEE 754 doubles [tests/igbinary_041.phpt]
        Check for double extremes [tests/igbinary_044.phpt]
        Correctly unserialize scalar refs. [tests/igbinary_046.phpt]
        Correctly unserialize multiple object refs. [tests/igbinary_046b.phpt]
        Correctly unserialize multiple array refs. [tests/igbinary_046c.phpt]
        Correctly unserialize multiple object refs and non-refs. [tests/igbinary_046d.phpt]
        Object test, __set not called for private attr in extended class [tests/igbinary_048b.phpt]
        Correctly unserialize multiple references in arrays [tests/igbinary_049.phpt]
        Correctly unserialize multiple references in objects [tests/igbinary_049b.phpt]
        Correctly unserialize cyclic object references [tests/igbinary_050.phpt]
        Object test, __wakeup (With multiple references) [tests/igbinary_051.phpt]
        Object Serializable interface can be serialized in references [tests/igbinary_052.phpt]
        __wakeup can modify properties without affecting other objects [tests/igbinary_053.phpt]
        __wakeup can add dynamic properties without affecting other objects [tests/igbinary_054.phpt]
        __wakeup can replace a copy of the object referring to the root node. [tests/igbinary_055.phpt]
        igbinary_unserialize causes segfault on 3rd call for objects with dynamic property [tests/igbinary_bug72134.phpt]
        Unserialize backwards compatible with v1. [tests/igbinary_unserialize_v1_compatible.phpt]
        =====================================================================

        You may have found a problem in PHP.
        This report can be automatically sent to the PHP QA team at
        http://qa.php.net/reports and http://news.php.net/php.qa.reports
        This gives us a better understanding of PHP's behavior.
        If you don't want to send the report immediately you can choose
        option "s" to save it.  You can then email it to qa-reports@lists.php.net later.
        Do you want to send this report now? [Yns]:

fyi, this is with

    gcc --version
        gcc (SUSE Linux) 6.2.1 20160830 [gcc-6-branch revision 239856]

--- a/src/php5/igbinary.c
+++ b/src/php5/igbinary.c
@@ -22,7 +22,7 @@
 #include "ext/standard/info.h"
 #include "ext/standard/php_var.h"
 
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 # include "ext/session/php_session.h"
 #endif /* HAVE_PHP_SESSION */
 
@@ -58,10 +58,10 @@
 #include <stddef.h>
 #include "hash.h"
 
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 /** Session serializer function prototypes. */
 PS_SERIALIZER_FUNCS(igbinary);
-#endif /* HAVE_PHP_SESSION */
+#endif /* HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION) */
 
 #if defined(HAVE_APC_SUPPORT) || defined(HAVE_APCU_SUPPORT)
 /** Apc serializer function prototypes */
@@ -246,7 +246,7 @@ zend_function_entry igbinary_functions[] = {
 #if ZEND_MODULE_API_NO >= 20050922
 static const zend_module_dep igbinary_module_deps[] = {
 	ZEND_MOD_REQUIRED("standard")
-#ifdef HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 	ZEND_MOD_REQUIRED("session")
 #endif
 #if defined(HAVE_APCU_SUPPORT)
@@ -307,7 +307,7 @@ PHP_MINIT_FUNCTION(igbinary) {
 	(void) module_number;
 	ZEND_INIT_MODULE_GLOBALS(igbinary, php_igbinary_init_globals, NULL);
 
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 	php_session_register_serializer("igbinary",
 		PS_SERIALIZER_ENCODE_NAME(igbinary),
 		PS_SERIALIZER_DECODE_NAME(igbinary));
@@ -355,7 +355,7 @@ PHP_MINFO_FUNCTION(igbinary) {
 #else
 	php_info_print_table_row(2, "igbinary APC serializer ABI", "no");
 #endif
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 	php_info_print_table_row(2, "igbinary session support", "yes");
 #else
 	php_info_print_table_row(2, "igbinary session support", "no");
@@ -496,7 +496,7 @@ PHP_FUNCTION(igbinary_serialize) {
 	RETVAL_STRINGL((char *)string, string_len, 0);
 }
 /* }}} */
-#ifdef HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 /* {{{ Serializer encode function */
 PS_SERIALIZER_ENCODE_FUNC(igbinary)
 {
@@ -601,7 +601,7 @@ PS_SERIALIZER_DECODE_FUNC(igbinary) {
 	return SUCCESS;
 }
 /* }}} */
-#endif /* HAVE_PHP_SESSION */
+#endif /* HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION) */
 
 #if defined(HAVE_APC_SUPPORT) || defined(HAVE_APCU_SUPPORT)
 /* {{{ apc_serialize function */

--- a/src/php7/igbinary.c
+++ b/src/php7/igbinary.c
@@ -21,9 +21,9 @@
 #include "ext/standard/info.h"
 #include "ext/standard/php_var.h"
 
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 # include "ext/session/php_session.h"
-#endif /* HAVE_PHP_SESSION */
+#endif
 
 #include "ext/standard/php_incomplete_class.h"
 
@@ -49,10 +49,10 @@
 #include "hash.h"
 #include "hash_ptr.h"
 
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 /** Session serializer function prototypes. */
 PS_SERIALIZER_FUNCS(igbinary);
-#endif /* HAVE_PHP_SESSION */
+#endif /* HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION) */
 
 #if defined(HAVE_APCU_SUPPORT)
 /** Apc serializer function prototypes */
@@ -248,7 +248,7 @@ zend_function_entry igbinary_functions[] = {
 /* {{{ igbinary dependencies */
 static const zend_module_dep igbinary_module_deps[] = {
 	ZEND_MOD_REQUIRED("standard")
-#ifdef HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 	ZEND_MOD_REQUIRED("session")
 #endif
 #if defined(HAVE_APCU_SUPPORT)
@@ -306,7 +306,7 @@ PHP_MINIT_FUNCTION(igbinary) {
 	(void) module_number;
 	ZEND_INIT_MODULE_GLOBALS(igbinary, php_igbinary_init_globals, NULL);
 
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 	php_session_register_serializer("igbinary",
 		PS_SERIALIZER_ENCODE_NAME(igbinary),
 		PS_SERIALIZER_DECODE_NAME(igbinary));
@@ -353,7 +353,7 @@ PHP_MINFO_FUNCTION(igbinary) {
 #else
 	php_info_print_table_row(2, "igbinary APC serializer ABI", "no");
 #endif
-#if HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 	php_info_print_table_row(2, "igbinary session support", "yes");
 #else
 	php_info_print_table_row(2, "igbinary session support", "no");
@@ -557,7 +557,7 @@ PHP_FUNCTION(igbinary_serialize) {
 	efree(string);
 }
 /* }}} */
-#ifdef HAVE_PHP_SESSION
+#if HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION)
 /* {{{ Serializer encode function */
 PS_SERIALIZER_ENCODE_FUNC(igbinary)
 {
@@ -649,7 +649,7 @@ PS_SERIALIZER_DECODE_FUNC(igbinary) {
 	return SUCCESS;
 }
 /* }}} */
-#endif /* HAVE_PHP_SESSION */
+#endif /* HAVE_PHP_SESSION && !defined(COMPILE_DL_SESSION) */
 
 #if defined(HAVE_APCU_SUPPORT)
 /* {{{ apc_serialize function */

